### PR TITLE
Update most dependencies

### PR DESCRIPTION
**Pull request by wb9688**
#### What is it?
- [ ] Bug fix (user facing)
- [ ] Feature (user facing)
- [x] Code base improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
Update all dependencies except ExoPlayer. Most changes come from upgrading to RxJava 3, which unfortunately requires upgrading to an alpha version of Room, and Ktlint rules changes.

Note how I've added 3 dependencies: `androidx.documentfile:documentfile` and `androidx.localbroadcastmanager:localbroadcastmanager` were previously part of some other AndroidX library, while `com.squareup.leakcanary:plumber-android` is a new feature from LeakCanary that fixes known memory leaks in the Android framework.

I did not upgrade ExoPlayer, as we probably want to use certain new features like audio offloading, `Media2` (replacement for `MediaSessionCompat`, which is now deprecated) support, and maybe toplevel playlist support. @Redirion: Do you want to do that in some PR?

#### Fixes the following issue(s)
Snyk complaining about vulnerabilities in some dependency. Edit: apparently it does not fix that for some reason.

#### Testing apk
[app-debug.apk.zip](https://github.com/TeamNewPipe/NewPipe/files/5377975/app-debug.apk.zip)

#### Agreement
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

### CODE REVIEW

**Comment by wb9688** [comment]
@opusforlife2: Could you test that everything still works? I'm mainly interested in if database import (please also test a database export from an *old* NewPipe version), subscription export, subscription import (especially its notification), and Markdown descriptions (as used by PeerTube) still work properly.

**Comment by opusforlife2** [comment]
Is it really alright to use an alpha version of a dependency? I remember F-Droid builds fail because alpha/beta versions of some Android build-related software aren't open source.

**Comment by wb9688** [comment]
@opusforlife2: Yes, the alpha versions of the AndroidX libraries are actually open-source (unlike the beta versions of the Android SDK itself), the following is the commits that were added in this alpha version compared to the latest non-alpha version: https://android.googlesource.com/platform/frameworks/support/+log/6880f4a443ed0d4e88e15c18044fbfc644c766e2..9f60cc700129e30cee9df020005c317fb39d32ec/room

**Comment by opusforlife2** [comment]
- Import database from 0.20.0 👍
- Import database from 0.19.8 👍
- Subscription export 👍
- Subscription import 👍
- Markdown in descriptions 👍

**Comment by Redirion** [comment]
@Redirion: Do you want to do that in some PR?

I am currently short of spare time :/ Maybe in 1-2 weeks.

**Comment by Stypox** [comment]
@wb9688 could you also update Kotlin to 1.4.10?

**Comment by wb9688** [comment]
@Stypox: Already did it at https://github.com/TeamNewPipe/NewPipe/pull/4508/files#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7R4 ;)

However I have to rebase this PR because of disabling Ktlint for now :/

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Fri May 01 19:39:41 CEST 2020
+#Thu Oct 15 11:41:05 CEST 2020
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-all.zip
```
Comment (Stypox) [inline gradle/wrapper/gradle-wrapper.properties:6]: Why `6.5`? Use `6.7` directly, I'd say, so that we don't have to upgrade again

**Review by Stypox — CHANGES_REQUESTED** [review]

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (wb9688) [reply gradle/wrapper/gradle-wrapper.properties:6]: No, this is the version suggested by Android Studio. AGP is usually a bit behind on Gradle versions.

**Comment by Stypox** [comment]
@Redirion upgrading ExoPlayer will hopefully fix this: in the middle of an mp4 stream the player suddenly says "Could not play this stream". From that point on trying to play the same cached stream will fail with the same error even if the playback is restarted. This error is given through logcat:
```
E/ExoPlayerImplInternal: Source error
      com.google.android.exoplayer2.ParserException: Invalid NAL length
        at com.google.android.exoplayer2.extractor.mp4.Mp4Extractor.readSample(Mp4Extractor.java:542)
        at com.google.android.exoplayer2.extractor.mp4.Mp4Extractor.read(Mp4Extractor.java:192)
        at com.google.android.exoplayer2.source.ProgressiveMediaPeriod$ExtractingLoadable.load(ProgressiveMediaPeriod.java:988)
        at com.google.android.exoplayer2.upstream.Loader$LoadTask.run(Loader.java:415)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:776)
```
By looking up on exoplayer's repo I found out this https://github.com/google/ExoPlayer/issues/7698, which is probably fixed by https://github.com/google/ExoPlayer/commit/f29af879c0458cc429af7c89f157603e2eb819f1, so we should be good

**Comment by opusforlife2** [comment]
@Stypox I think I asked at least one of those to test with WebM as well? I don't remember clearly.

**Pull request closed without merging**